src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -391,10 +435,16 @@ struct
 
   let join a b =
     let join_d ad bd =
-      (*use copy of ad because result is later saved in there*)
-      let ad = Array.copy ad in
       (*This is the table which is later grouped*)
-      let table = BatList.map2i (fun i (ai, aj) (bi,bj) -> (i, Z.(aj - bj), (ai, aj), (bi,bj))) (Array.to_list ad) (Array.to_list bd) in
+      let joinfunction = fun lhs rhs1 rhs2 -> match rhs1, rhs2 with 
```

Comment:
Any particular reason for `fun` instead of having the arguments on the left side?

Reply:
The reason is me being a fool in ocaml